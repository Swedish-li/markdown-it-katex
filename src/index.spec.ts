import mdk from './index'
import MarkdownIt from 'markdown-it'

const md = MarkdownIt().use(mdk)
describe('markdown katex render', () => {
  it('Simple inline math', () => {
    const output = md.render(
      `$1+1 = 2$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Simple block math', () => {
    const output = md.render(
      `$$1+1 = 2$$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('No whitespace before and after is fine', () => {
    const output = md.render(
      `foo$1+1 = 2$bar
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Even when it starts with a negative sign', () => {
    const output = md.render(
      `foo$-1+1 = 2$bar
`
    )

    expect(output).toMatchSnapshot()
  })

  it("Shouldn't render empty content", () => {
    const output = md.render(
      `aaa $$ bbb
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Should require a closing delimiter', () => {
    const output = md.render(
      `aaa $5.99 bbb
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Paragraph break in inline math is not allowed', () => {
    const output = md.render(
      `foo $1+1

= 2$ bar
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Inline math with apparent markup should not be processed', () => {
    const output = md.render(
      `foo $1 *i* 1$ bar
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Block math can be indented up to 3 spaces', () => {
    const output = md.render(
      `   $$
 1+1 = 2
 $$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('But 4 means a code block', () => {
    const output = md.render(
      `    $$
  1+1 = 2
  $$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Multiline inline math', () => {
    const output = md.render(
      `foo $1 + 1
= 2$ bar
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Multiline display math', () => {
    const output = md.render(
      `$$

1
+ 1

= 2

$$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Text can immediately follow inline math', () => {
    const output = md.render(
      `$n$-th order
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Display math self-closes at the end of document', () => {
    const output = md.render(
      `$$
1+1 = 2
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Display and inline math can appear in lists', () => {
    const output = md.render(
      `* $1+1 = 2$
* $$
1+1 = 2
$$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Display math can be written in one line', () => {
    const output = md.render(
      `$$1+1 = 2$$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Or on multiple lines with expression starting and ending on delimited lines', () => {
    const output = md.render(
      `$$[
[1, 2]
[3, 4]
]$$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Escaped delimiters should not render math', () => {
    const output = md.render(
      `Foo \$1$ bar
\$\$
1
\$\$
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Numbers can not follow closing inline math', () => {
    const output = md.render(
      `Thus, $20,000 and USD$30,000 won't parse as math.
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Require non whitespace to right of opening inline math', () => {
    const output = md.render(
      `For some Europeans, it is 2$ for a can of soda, not 1$.
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Require non whitespace to left of closing inline math.', () => {
    const output = md.render(
      `I will give you $20 today, if you give me more $ tomorrow.
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Inline blockmath is not (currently) registered.', () => {
    const output = md.render(
      `It's well know that $$1 + 1 = 3$$ for sufficiently large 1.
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Escaped delimiters in math mode', () => {
    const output = md.render(
      `Money adds: $\$X + \$Y = \$Z$.
`
    )

    expect(output).toMatchSnapshot()
  })

  it('Multiple escaped delimiters in math module', () => {
    const output = md.render(
      `Weird-o: $\displaystyle{\begin{pmatrix} \$ & 1\\\$ \end{pmatrix}}$.
`
    )

    expect(output).toMatchSnapshot()
  })
})
