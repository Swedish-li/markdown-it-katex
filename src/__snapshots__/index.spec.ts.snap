// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown katex render Block math can be indented up to 3 spaces 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1+1 = 2

</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span></span></p>"
`;

exports[`markdown katex render But 4 means a code block 1`] = `
"<pre><code>$$
1+1 = 2
$$
</code></pre>
"
`;

exports[`markdown katex render Display and inline math can appear in lists 1`] = `
"<ul>
<li><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1+1 = 2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span></li>
<li>
<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow></mrow><annotation encoding=\\"application/x-tex\\">
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"></span></span></span></p></li>
</ul>
<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow></mrow><annotation encoding=\\"application/x-tex\\">
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"></span></span></span></p>"
`;

exports[`markdown katex render Display math can be written in one line 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1+1 = 2

</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span></span></p>"
`;

exports[`markdown katex render Display math self-closes at the end of document 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1+1 = 2

</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span></span></p>"
`;

exports[`markdown katex render Escaped delimiters in math mode 1`] = `
"<p>Money adds: <span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi mathvariant=\\"normal\\">$</mi><mi>X</mi><mo>+</mo><mi mathvariant=\\"normal\\">$</mi><mi>Y</mi><mo>=</mo><mi mathvariant=\\"normal\\">$</mi><mi>Z</mi></mrow><annotation encoding=\\"application/x-tex\\">\\\\$X + \\\\$Y = \\\\$Z</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.83333em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">$</span><span class=\\"mord mathnormal\\" style=\\"margin-right:0.07847em;\\">X</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.80556em;vertical-align:-0.05556em;\\"></span><span class=\\"mord\\">$</span><span class=\\"mord mathnormal\\" style=\\"margin-right:0.22222em;\\">Y</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.80556em;vertical-align:-0.05556em;\\"></span><span class=\\"mord\\">$</span><span class=\\"mord mathnormal\\" style=\\"margin-right:0.07153em;\\">Z</span></span></span></span>.</p>
"
`;

exports[`markdown katex render Escaped delimiters should not render math 1`] = `
"<p>Foo $1$ bar
$$
1
$$</p>
"
`;

exports[`markdown katex render Even when it starts with a negative sign 1`] = `
"<p>foo<span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mo>−</mo><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">-1+1 = 2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">−</span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span>bar</p>
"
`;

exports[`markdown katex render Inline blockmath is not (currently) registered. 1`] = `
"<p>It's well know that $$1 + 1 = 3$$ for sufficiently large 1.</p>
"
`;

exports[`markdown katex render Inline math with apparent markup should not be processed 1`] = `
"<p>foo <span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mn>1</mn><mo>∗</mo><mi>i</mi><mo>∗</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">1 *i* 1</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">∗</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.65952em;vertical-align:0em;\\"></span><span class=\\"mord mathnormal\\">i</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">∗</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span></span></span></span> bar</p>
"
`;

exports[`markdown katex render Multiline display math 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">
1
+ 1

= 2


</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span></span></p>"
`;

exports[`markdown katex render Multiline inline math 1`] = `
"<p>foo <span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1 + 1
= 2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span> bar</p>
"
`;

exports[`markdown katex render Multiple escaped delimiters in math module 1`] = `
"<p>Weird-o: <span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mstyle scriptlevel=\\"0\\" displaystyle=\\"true\\"><mrow><mo fence=\\"true\\">(</mo><mtable rowspacing=\\"0.15999999999999992em\\" columnspacing=\\"1em\\"><mtr><mtd><mstyle scriptlevel=\\"0\\" displaystyle=\\"false\\"><mi mathvariant=\\"normal\\">$</mi></mstyle></mtd><mtd><mstyle scriptlevel=\\"0\\" displaystyle=\\"false\\"><mn>1</mn></mstyle></mtd></mtr><mtr><mtd><mstyle scriptlevel=\\"0\\" displaystyle=\\"false\\"><mi mathvariant=\\"normal\\">$</mi></mstyle></mtd></mtr></mtable><mo fence=\\"true\\">)</mo></mrow></mstyle></mrow><annotation encoding=\\"application/x-tex\\">\\\\displaystyle{\\\\begin{pmatrix} \\\\$ &amp; 1\\\\\\\\\\\\$ \\\\end{pmatrix}}</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:2.40003em;vertical-align:-0.95003em;\\"></span><span class=\\"mord\\"><span class=\\"minner\\"><span class=\\"mopen delimcenter\\" style=\\"top:0em;\\"><span class=\\"delimsizing size3\\">(</span></span><span class=\\"mord\\"><span class=\\"mtable\\"><span class=\\"col-align-c\\"><span class=\\"vlist-t vlist-t2\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:1.45em;\\"><span style=\\"top:-3.61em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\">$</span></span></span><span style=\\"top:-2.4099999999999997em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\">$</span></span></span></span><span class=\\"vlist-s\\">​</span></span><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.9500000000000004em;\\"><span></span></span></span></span></span><span class=\\"arraycolsep\\" style=\\"width:0.5em;\\"></span><span class=\\"arraycolsep\\" style=\\"width:0.5em;\\"></span><span class=\\"col-align-c\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:1.45em;\\"><span style=\\"top:-3.61em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\">1</span></span></span></span></span></span></span></span></span><span class=\\"mclose delimcenter\\" style=\\"top:0em;\\"><span class=\\"delimsizing size3\\">)</span></span></span></span></span></span></span>.</p>
"
`;

exports[`markdown katex render No whitespace before and after is fine 1`] = `
"<p>foo<span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1+1 = 2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span>bar</p>
"
`;

exports[`markdown katex render Numbers can not follow closing inline math 1`] = `
"<p>Thus, $20,000 and USD$30,000 won't parse as math.</p>
"
`;

exports[`markdown katex render Or on multiple lines with expression starting and ending on delimited lines 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mo stretchy=\\"false\\">[</mo><mo stretchy=\\"false\\">[</mo><mn>1</mn><mo separator=\\"true\\">,</mo><mn>2</mn><mo stretchy=\\"false\\">]</mo><mo stretchy=\\"false\\">[</mo><mn>3</mn><mo separator=\\"true\\">,</mo><mn>4</mn><mo stretchy=\\"false\\">]</mo><mo stretchy=\\"false\\">]</mo></mrow><annotation encoding=\\"application/x-tex\\">[
[1, 2]
[3, 4]
]
</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:1em;vertical-align:-0.25em;\\"></span><span class=\\"mopen\\">[</span><span class=\\"mopen\\">[</span><span class=\\"mord\\">1</span><span class=\\"mpunct\\">,</span><span class=\\"mspace\\" style=\\"margin-right:0.16666666666666666em;\\"></span><span class=\\"mord\\">2</span><span class=\\"mclose\\">]</span><span class=\\"mopen\\">[</span><span class=\\"mord\\">3</span><span class=\\"mpunct\\">,</span><span class=\\"mspace\\" style=\\"margin-right:0.16666666666666666em;\\"></span><span class=\\"mord\\">4</span><span class=\\"mclose\\">]</span><span class=\\"mclose\\">]</span></span></span></span></span></p>"
`;

exports[`markdown katex render Paragraph break in inline math is not allowed 1`] = `
"<p>foo $1+1</p>
<p>= 2$ bar</p>
"
`;

exports[`markdown katex render Require non whitespace to left of closing inline math. 1`] = `
"<p>I will give you $20 today, if you give me more $ tomorrow.</p>
"
`;

exports[`markdown katex render Require non whitespace to right of opening inline math 1`] = `
"<p>For some Europeans, it is 2$ for a can of soda, not 1$.</p>
"
`;

exports[`markdown katex render Should require a closing delimiter 1`] = `
"<p>aaa $5.99 bbb</p>
"
`;

exports[`markdown katex render Shouldn't render empty content 1`] = `
"<p>aaa $$ bbb</p>
"
`;

exports[`markdown katex render Simple block math 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1+1 = 2

</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span></span></p>"
`;

exports[`markdown katex render Simple inline math 1`] = `
"<p><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mn>1</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>2</mn></mrow><annotation encoding=\\"application/x-tex\\">1+1 = 2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.72777em;vertical-align:-0.08333em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">2</span></span></span></span></p>
"
`;

exports[`markdown katex render Text can immediately follow inline math 1`] = `
"<p><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>n</mi></mrow><annotation encoding=\\"application/x-tex\\">n</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.43056em;vertical-align:0em;\\"></span><span class=\\"mord mathnormal\\">n</span></span></span></span>-th order</p>
"
`;
